internal/npm_install/generate_build_file.js

```diff
@@ -527,13 +527,14 @@ function parsePackage(p) {
 
   // For root packages, transform the pkg.bin entries
   // into a new Map called _executables
+  // NOTE: we do this only non-empty bin entries
   pkg._executables = new Map();
   if (!pkg._isNested) {
     if (Array.isArray(pkg.bin)) {
       // should not happen, but ignore it if present
-    } else if (typeof pkg.bin === 'string') {
+    } else if (typeof pkg.bin === 'string' && pkg.bin !== '') {
       pkg._executables.set(pkg._dir, cleanupBinPath(pkg.bin));
-    } else if (typeof pkg.bin === 'object') {
+    } else if (typeof pkg.bin === 'object' && Object.keys(pkg.bin).length > 0) {
```

Comment:
this doesn't make sense:
- if it's `bin: {}` then the loop on the next line has zero iterations, so the added conditional doesn't do anything
- it doesn't cover the case of `bin: {"thing": "}` which I thought was the point of the PR?

do we have unit tests for this file?

Reply:
Right. I'll fix that and add some tests.